pkg/cloud-init/cloud-init.go

```diff
@@ -120,6 +130,40 @@ func cloudInitUUIDFromVMI(vmi *v1.VirtualMachineInstance) string {
 	return string(vmi.Spec.Domain.Firmware.UUID)
 }
 
+func generateHostname(vmi *v1.VirtualMachineInstance, userData string) (string, error) {
```

Comment:
Nit: this is not really "generting" a hostname, it's just extracting it from userData/VMI

Reply:
Yup, I agree. Done :+1: